src/edu/csus/ecs/pc2/clics/API202306/ScoreboardService.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Webservice to handle scoreboard requests
+ *
+ * @author ICPC
+ */
+@Path("/contests/{contestId}/scoreboard")
+@Produces(MediaType.APPLICATION_JSON)
+@Provider
+@Singleton
+public class ScoreboardService implements Feature {
+
+    private IInternalContest model;
+    private IInternalController controller;
+
+    public ScoreboardService(IInternalContest inContest, IInternalController inController) {
+        super();
+        this.model = inContest;
+        this.controller = inController;
+    }
+
+    /**
+     * This method returns a representation of the current contest scoreboard in JSON format.
+     * The return JSON is in the format defined by the CLICS 2023-06 spec.
+     *
+     * @param servletRequest
+     * @param sc
+     * @param contestId
+     * @param group_id - optional group id query param
+     * @return {@link Response} object containing the JSON scoreboard
+     */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getScoreboard(@Context HttpServletRequest servletRequest, @Context SecurityContext sc, @PathParam("contestId") String contestId,
+            @QueryParam("group_id") String group_id, @QueryParam("division") String division) {
+
+        // check contest id
+        if(contestId.equals(model.getContestIdentifier()) == true) {
+            ContestTime contestTime = model.getContestTime();
+            // verify contest has started or user is special
+            if (contestTime.getElapsedMS() > 0 ||
+                ((sc.isUserInRole(WebServer.WEBAPI_ROLE_ADMIN) ||
+                  sc.isUserInRole(WebServer.WEBAPI_ROLE_ANALYST) ||
+                  sc.isUserInRole(WebServer.WEBAPI_ROLE_JUDGE)))) {
+
+                Group specificGroup = null;
+                Integer divNumber = null;
+
+                // if a specific group was requested, let's look for that so we can pass it to the standings routine
+                if(!StringUtilities.isEmpty(group_id)) {
+                    for(Group group: model.getGroups()) {
+                        if (group.isDisplayOnScoreboard() && IJSONTool.getGroupId(group).equals(group_id)) {
+                            specificGroup = group;
+                            break;
+                        }
+                    }
+                    if(specificGroup == null) {
+                        return Response.status(Response.Status.NOT_FOUND).build();
+                    }
+                }
+
+                // see if a division number was specified - this is a PC2 extension(!!) and is NOT part of the CLICS API, but,
+                // we are allowed to add things.
+                if(!StringUtilities.isEmpty(division)) {
+                    try {
+                        divNumber = Integer.parseInt(division);
+                    } catch(Exception e) {
+                        // Bad division specified
+                        return Response.status(Response.Status.NOT_FOUND).build();
```

Comment:
It seems like "BAD_REQUEST" would be more appropriate here...

Reply:
Yep.  A number is required here, so it is a syntax error.  Fixed in next push.